Sacred Protocol

SacredProtocol opened the pull request:
### 💪 Project Abstract

Sacred Protocol is a decentralized tipping primitive that allows creators to receive support from fans across borders, platforms, and currencies. We aim to break open the creator economy for the 80–90% of audiences currently unable to tip due to geographic, platform, and payment limitations.

We want to enable social tipping for the Polkadot ecosystem on any social media platform. We want to enable audiences and creators within the community to be able to reward, engage and encourage each other in real time using the Polkadot blockchain to empower settlement and security. 


### ✅ Application Checklist

- [x] The [application template](https://github.com/Polkadot-Fast-Grants/apply/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read and followed the [program guidelines](https://github.com/Polkadot-Fast-Grants/apply#guidelines).
- [x] I understand that grants are capped at **$10,000 USD per application** with a maximum of **$15,000 USD per project** in exceptional cases.
- [x] I understand that each milestone is capped at **$5,000 USD** and must be delivered within 3 months.
- [x] I understand that payments are made only after successful milestone delivery.
- [x] I have clearly explained what the funding is going towards (e.g. 30 hours of a full time employee at $X / hour).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.

dimdomsum commented:
Thanks team! Your application is now in review. Please could you provide more info on how your project will add long-term value to the Polkadot ecosystem and provide a more detailed budget breakdown.

SacredProtocol commented on applications/sacred-protocol.md:
```diff
@@ -0,0 +1,201 @@
+
+## 🔮 Future Plans
+
+- Convert core business logic into Smart Contracts (ie. Escrow Mechanisms)
```
**Over the next iterations we have in store, Sacred’s north‑star is to make DOT (and parachain‑native USDC) the default “thank‑you” currency of the internet. Every tip will travel via XCM, hopping transparently between the Relay Chain and any integrated parachain, so creators and fans experience a single friction‑free wallet while Polkadot captures the fee flow. Once this internal flywheel is proven, we will layer in bridge adapters (e.g. Snowbridge, Wormhole) that convert ETH, SOL, or other L1 liquidity into DOT before it enters Sacred, importing external value—and new users—into the Polkadot economy while keeping settlement trust‑minimised.**

- Convert core business logic into ink! smart contracts (e.g. escrow/dispute‑resolution modules) to remove all off‑chain trust assumptions.

- Expand support for native tips on additional Polkadot wallets and parachains, ensuring one‑click UX across the ecosystem.

- Launch a Creator Analytics Dashboard so creators can track tipping history, top fans, and cross‑parachain revenue.

- Partner with mainstream content platforms (e.g. Kick, BlueSky) to embed Polkadot‑powered tipping in their creator tooling.

- Pursue additional grants or venture support to scale transaction volume and liquidity incentives.

- Expand integrations with the official Polkadot App (referendum 1573) to drive greater on‑chain adoption and visibility.

SacredProtocol commented:
Hi @oshakarishvili,

I appreciate you and the team reviewing our proposal. We've included our responses below and adjusted the budget on the application item itself.

> 1. When a fan clicks “Tip”, do DOT/USDC go straight into a permissionless contract, or does your backend briefly control the funds? What happens to tips that remain unclaimed?

Answer:
Contracts are what we we are working up to. For now as we get the user flow right, the funds go to a wallet controlled by the backend server logic. The server maintains a record of who sent the funds (wallet address) and who is the recipient (social handle). At this stage if a tip is unclaimed, the sender can “revoke” the tip and the funds will be returned to the sender’s wallet address.

In the future, we plan to have unclaimed tips follow a decay formula and distributed to active users in the Sacred ecosystem and/or Sacred token holders to incentivize tipping, however this is out of scope of this grant application. See our litepaper for the future vision: https://www.sacredprotocol.com/litepaper .

> 2. Would you be willing to release the Polkadot escrow/claim contracts under MIT or Apache-2.0 (not just the CLI)?

Answer:
When we publish and deploy any contracts it will be open source. At this stage there is no smart contract code. We plan to implement an escrow contract with oracle-based logic as part of our decentralization roadmap – for now this is outside of the scope of the current proposal.

> 3. (T)witter/X and Twitch sometimes throttle tipping bots. Have you confirmed that your current implementation complies with their latest ToS, and do you have a fallback plan if API access is limited?

Answer:
We carefully review and comply with each platform’s Terms of Service and make sure that we operate within the rules and are compliant. The app does not use any “bot” behavior—all interactions are user-initiated. We use official platform APIs for OAuth verification to confirm tip recipients own their social accounts. For other data and content lookups, we leverage both official APIs and third-party data aggregators.

> 4. Fast-Grant requests are capped at $ 10k. Can you revise the milestones to fit that limit (perhaps phasing marketing/creator payouts against usage KPIs)?

Answer: Budget items on proposal updated

SacredProtocol commented:
@dimdomsum @oshakarishvili 
Hello Polkadot Fast Grants team! We have addressed your feedback and made changes to the milestones and deliverables. We believe this better aligns with the decentralization aspects to help contribute to the Polkadot ecosystem growth better.

Please review the Milestones and Deliverables, along with the budget breakdown, with 3 core milestones over a 2 month timeline.

oshakarishvili commented:
For deliverable reporting, just follow the standard process: push your code/docs to your repo and submit the milestone PR in the delivery repo. You can post updates here as well. A group chat isn’t necessary, but feel free to tag us either here or on Telegram support channel anytime if you have blockers or questions
